### PR TITLE
[Dashboard] Add utility functions for actor and memory APIs

### PR DESCRIPTION
**Pull request by mfitton**
## Why are these changes needed?
There are several utility classes / functions leaned upon by the Memory and Logical View APIs. This introduces the utility modules.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

Memory utils is unit tested. There are integration tests in the PR with the meat of the memory view / logical view logic.

### CODE REVIEW

File: dashboard/memory_utils.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+        elif sort_by_type:
+            self._sort_by(sort_by_type)
+
+    def setup(self, group_by_type: GroupByType, sort_by_type: SortingType):
+        """Setup memory table.
+
+        This will sort entries first and gruop them after.
+        Sort order will be still kept.
+        """
+        self._sort_by(sort_by_type)._group_by(group_by_type)
```
Comment (fyrestone) [inline dashboard/memory_utils.py:184]: Is the `self._sort_by(...)._group_by(...)` equals with `self._group_by(...)._sort_by(...)`?

File: dashboard/memory_utils.py [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+        random_bits = object_ref_hex[:TASKID_RANDOM_BITS_SIZE]
+        actor_random_bits = object_ref_hex[TASKID_RANDOM_BITS_SIZE:
+                                           TASKID_RANDOM_BITS_SIZE +
+                                           ACTORID_RANDOM_BITS_SIZE]
+        if (random_bits == "f" * 16 and not actor_random_bits == "f" * 8):
+            return True
+        else:
+            return False
+
+    def __dict__(self):
```
[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/memory_utils.py:138]: This is a magic method.

[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/memory_utils.py:138]: I think as_dict is also fine though.

File: dashboard/actor_utils.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                .get("pythonFunctionDescriptor", {})\
+                .get("className", "Unknown actor class")
+
+
+def _group_actors_by_python_class(actors):
+    groups = defaultdict(list)
+    for actor in actors.values():
+        actor_class = actor["actorClass"]
+        groups[actor_class].append(actor)
+    return dict(groups)
```
Comment (rkooo567) [inline dashboard/actor_utils.py:38]: Isn't groups already dict?

**Review by rkooo567 — COMMENTED** [review]
This PR doesn't actually seem to add memory dashboard functionalities to the new dashboard?

File: dashboard/memory_utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fyrestone) [reply dashboard/memory_utils.py:138]: But, I can't find the official document for the `__dict__` method. I know there is a `__dict__` special attribute, it's not a method.

[the remark this answers is not in the document]
Comment (rkooo567) [reply dashboard/memory_utils.py:138]: Ah, you are right. I've searched about it, and it was not the magic method :o 

I didn't know about this so far... We should change to `as_dict`. Thanks for caching this!

[the remark this answers is not in the document]
Comment (mfitton) [reply dashboard/memory_utils.py:138]: Yeah, __dict__ returns the Python object's attributes as a dict. Good catch, I'll rename it.

File: dashboard/actor_utils.py (hunk 3 above)
Comment (mfitton) [reply dashboard/actor_utils.py:38]: No groups is a defaultdict.

**Comment by mfitton** [comment]
@rkooo567 correct, this PR just adds some utility files that get used by the logical view and memory view in another PR.

File: dashboard/memory_utils.py (hunk 1 above)
Comment (mfitton) [reply dashboard/memory_utils.py:183]: It is not equal, which I agree is somewhat confusing. That said, it is sorted before grouped in all code paths.

**Review by rkooo567 — APPROVED** [review]

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**